src/util/stft.rs

```diff
@@ -245,6 +247,40 @@ impl<const NUM_SIDECHAIN_INPUTS: usize> StftHelper<NUM_SIDECHAIN_INPUTS> {
         self.current_pos = 0;
     }
 
+    /// Change the current padding. This will clear the buffers, causing the next block to output
+    /// silence.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if `padding > max_padding`.
+    pub fn set_padding(&mut self, padding: usize) {
+        assert!(padding <= self.padding_buffers[0].capacity());
+
+        self.padding = padding;
+        let block_size = self.main_input_ring_buffers[0].len();
+        for main_ring_buffer in &mut self.main_input_ring_buffers {
+            main_ring_buffer.fill(0.0);
+        }
+        for main_ring_buffer in &mut self.main_output_ring_buffers {
+            main_ring_buffer.fill(0.0);
+        }
+        for sidechain_ring_buffers in &mut self.sidechain_ring_buffers {
+            for sidechain_ring_buffer in sidechain_ring_buffers {
+                sidechain_ring_buffer.fill(0.0);
+            }
+        }
+        self.scratch_buffer.resize(block_size + self.padding, 0.0);
+        self.scratch_buffer.fill(0.0);
+
+        // For consistency's sake we'll also clear this here
+        for padding_buffer in &mut self.padding_buffers {
+            padding_buffer.resize(self.padding, 0.0);
+            padding_buffer.fill(0.0);
+        }
+
+        self.current_pos = 0;
+    }
```

Comment:
All of this resizing can be moved into a single internal function to avoid consistency issues later. I can do that myself after merging though.